client/context/EditorContext.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { editObjectCatalogUrl, getObjectModelsDatastreamsUrl } from "../util/routes";
+import { useFetchContext } from "./FetchContext";
+
+/**
+ * Pass a shared entity to react components,
+ * specifically a way to make api requests.
+ */
+const editorContextParams = {
+    modelsCatalog: {},
+    currentPid: null,
+    currentModels: [],
+    currentDatastreams: [],
+    currentDatastream: null,
```

Comment:
Having variables called currentDatastream and currentDatastreams seems potentially confusing. If currentDatastreams is all the datastreams present on the current object, and currentDatastream is the one the user is interacting with presently, maybe renaming currentDatastream to activeDatastream would better differentiate the naming.

Reply:
I'll make that change.